cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/DefaultPluginConfigurer.java

```diff
@@ -133,29 +142,24 @@ public void addPlugins(Map<String, Plugin> plugins) {
       pluginEntry = artifactRepository.findPlugin(artifactId, pluginType, pluginName, selector);
     } catch (IOException e) {
       throw Throwables.propagate(e);
-    } catch (PluginNotExistsException e) {
-      throw new IllegalArgumentException(String.format("Plugin of type %s, name %s could not be found",
-                                                       pluginType, pluginName), e);
     }
 
-    if (pluginEntry != null) {
-      // Just verify if all required properties are provided.
-      // No type checking is done for now.
-      for (PluginPropertyField field : pluginEntry.getValue().getProperties().values()) {
-        Preconditions.checkArgument(!field.isRequired() || properties.getProperties().containsKey(field.getName()),
-                                    "Required property '%s' missing for plugin of type %s, name %s.",
-                                    field.getName(), pluginType, pluginName);
-      }
+    // Just verify if all required properties are provided.
+    // No type checking is done for now.
+    for (PluginPropertyField field : pluginEntry.getValue().getProperties().values()) {
+      Preconditions.checkArgument(!field.isRequired() ||
+                                    (properties != null && properties.getProperties().containsKey(field.getName())),
+                                  "Required property '%s' missing for plugin of type %s, name %s.",
+                                  field.getName(), pluginType, pluginName);
     }
-    return pluginEntry;
-  }
 
-  /**
-   * Register the given plugin in this configurer.
-   */
-  private void registerPlugin(String pluginId, ArtifactDescriptor artifactDescriptor, PluginClass pluginClass,
-                              PluginProperties properties) {
-    plugins.put(pluginId, new Plugin(artifactDescriptor.getArtifactId(), artifactDescriptor.getLocation().toURI(),
-                                     pluginClass, properties));
+    ArtifactId artifactId = pluginEntry.getKey().getArtifactId();
+    File destFile = new File(pluginInstantiator.getBaseDir(), ArtifactFile.getFileName(artifactId));
```

Comment:
Maybe better to move this logic into `PluginInstantiator` than exposing the baseDir. You can have `PluginInstantiator.addArtifact(Location artifactLocation)`.